table/block_based/block_based_table_builder.cc

```diff
@@ -471,6 +482,13 @@ struct BlockBasedTableBuilder::Rep {
   ~Rep() {}
 };
 
+// offset will only be updated by only one thread in WriteRawBlock at any time,
+// disable TSAN.
+DISABLE_TSAN
```

Comment:
In general we prefer rewriting the code so that the sanitizer to understand than disabling the check.